pkg/kubelet/status/generate.go

```diff
@@ -69,6 +71,15 @@ func GenerateContainersReadyCondition(spec *v1.PodSpec, containerStatuses []v1.C
 		}
 	}
 
+	// If the pod phase is failed, explicitly set the ready condition to false for containers since they may be in progress of terminating.
+	if podPhase == v1.PodFailed {
```

Comment:
Why is this not also checking `PodSucceeded` ?  Should we leave a comment?

Reply:
And why does this not call `generateContainersReadyConditionForTerminalPhase(podPhase)` ?

Reply:
> Why is this not also checking PodSucceeded ? Should we leave a comment?

PodSucceded is already handled above on line 66, see [here](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/status/generate.go#L64-L70)

> And why does this not call generateContainersReadyConditionForTerminalPhase(podPhase) ?

I created that helper function to be used in `status_manager.go` to be able to handle either phase. I returned the condition directly inline here, since it's what done in the succeeded case above, but I agree with you, we can switch it to use the helper function as well.

Reply:
I updated, based on your feedback and reused `generateContainersReadyConditionForTerminalPhase` for both succeeded and failed cases, so the condition generation logic is in one place.